sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/service/write/DocumentWriteServiceTest.java

```diff
@@ -0,0 +1,1192 @@
+  @Inject DataStoreProperties dataStoreProperties;
+  @Inject DocumentProperties documentProperties;
+  @InjectMock TimeSource timeSource;
+
+  String keyspaceName;
+  String tableName;
+  Batch.Type expectedBatchType;
+  String documentId;
+  long timestamp;
+  ExecutionContext context;
+
+  @BeforeEach
+  public void init() {
+    keyspaceName = schemaProvider.getKeyspace().getName();
+    tableName = schemaProvider.getTable().getName();
+    expectedBatchType =
+        dataStoreProperties.loggedBatchesEnabled() ? Batch.Type.LOGGED : Batch.Type.UNLOGGED;
+    documentId = RandomStringUtils.randomAlphanumeric(16);
+    timestamp = RandomUtils.nextLong();
+    when(timeSource.currentTimeMicros()).thenReturn(timestamp);
+    context = ExecutionContext.create(true);
+  }
+
+  @Nested
+  class WriteDocument {
+
+    @Test
+    public void happyPath() {
+      JsonShreddedRow row1 =
+          ImmutableJsonShreddedRow.builder()
+              .maxDepth(documentProperties.maxDepth())
+              .addPath("key1")
+              .stringValue("value1")
+              .build();
+      JsonShreddedRow row2 =
+          ImmutableJsonShreddedRow.builder()
+              .maxDepth(documentProperties.maxDepth())
+              .addPath("key2")
+              .addPath("nested")
+              .doubleValue(2.2d)
+              .build();
+      List<JsonShreddedRow> rows = Arrays.asList(row1, row2);
+
+      String insertCql =
+          String.format(
+              "INSERT INTO %s.%s "
+                  + "(key, p0, p1, p2, p3, leaf, text_value, dbl_value, bool_value) "
+                  + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?) USING TIMESTAMP ?",
+              keyspaceName, tableName);
+      ValidatingStargateBridge.QueryAssert row1QueryAssert =
+          withQuery(
+                  insertCql,
+                  Values.of(documentId),
+                  Values.of("key1"),
+                  Values.of(""),
+                  Values.of(""),
+                  Values.of(""),
+                  Values.of("key1"),
+                  Values.of("value1"),
+                  Values.NULL,
+                  Values.NULL,
+                  Values.of(timestamp))
+              .inBatch(expectedBatchType)
+              .returningNothing();
+      ValidatingStargateBridge.QueryAssert row2QueryAssert =
+          withQuery(
+                  insertCql,
+                  Values.of(documentId),
+                  Values.of("key2"),
+                  Values.of("nested"),
+                  Values.of(""),
+                  Values.of(""),
+                  Values.of("nested"),
+                  Values.NULL,
+                  Values.of(2.2d),
+                  Values.NULL,
+                  Values.of(timestamp))
+              .inBatch(expectedBatchType)
+              .returningNothing();
+
+      service
+          .writeDocument(keyspaceName, tableName, documentId, rows, null, context)
+          .await()
+          .atMost(ASYNC_WAIT_DURATION);
```

Comment:
Would propose to use the multiny assertion:

```
 service
          .writeDocument(keyspaceName, tableName, documentId, rows, null, context)
          .subscribe()
          .withSubscriber(UniAssertSubscriber.create())
          .awaitItem()
          .assertItem(...) // optional step, if needed
          .assertCompleted();
```

The we also don't need the `ASYNC_WAIT_DURATION`